src/common.rs

```diff
@@ -11,27 +11,82 @@ pub trait AsU32 {
 cfg_if::cfg_if! {
     if #[cfg(any(windows, target_os = "unknown", target_arch = "wasm32"))] {
         /// Process id.
-        pub type Pid = usize;
+        #[derive(Debug, Clone, Copy, Eq, Hash)]
+        pub struct Pid(usize);
 
-        impl AsU32 for Pid {
-            fn as_u32(&self) -> u32 {
-                *self as u32
+        impl From<u32> for Pid {
+            fn from(u32v: u32) -> Pid {
+                Pid(u32v as usize)
             }
         }
+
+        impl From<i32> for Pid {
```

Comment:
I'd rather not have this implementation. There doesn't seem to have much point in having `i32` conversions...